libs/providers/config-cat-web/src/lib/config-cat-web-provider.ts

```diff
@@ -137,13 +141,22 @@ export class ConfigCatWebProvider implements Provider {
 
     const configCatDefaultValue = flagType !== 'object' ? (defaultValue as SettingValue) : JSON.stringify(defaultValue);
 
-    const { value, ...evaluationData } = this._client
-      .snapshot()
-      .getValueDetails(flagKey, configCatDefaultValue, transformContext(context));
+    const snapshot = this._client.snapshot();
+
+    const { value, ...evaluationData } = snapshot.getValueDetails(
+      flagKey,
+      configCatDefaultValue,
+      transformContext(context),
+    );
+
+    if (!this._isProviderReady && snapshot.cacheState !== ClientCacheState.NoFlagData) {
+      // Ideally, we would check ConfigCat client's initialization state in its "background" polling loop.
+      // This is not possible at the moment, so as a workaround, we do the check on feature flag evaluation.
+      // There are plans to improve this situation, so let's revise this
+      // as soon as ConfigCat SDK implements the necessary event.
 
-    if (this._hasError && !evaluationData.errorMessage && !evaluationData.errorException) {
-      this._hasError = false;
-      this.events.emit(ProviderEvents.Ready);
+      this._isProviderReady = true;
+      setTimeout(() => this.events.emit(ProviderEvents.Ready), 0);
```

Comment:
Is there a reason to not emit this event right away? Also, this will likely trigger a re-render in the React SDK but it should no longer trigger a re-render loop.

Reply:
It's just that it doesn't feel right to invoke unknown user code during the evaluation of a feature flag, when normally that code would be invoked asynchronously, on the client's "background" polling loop. (As I mentioned in the related issue, the necessary event is not emitted by ConfigCat client at the moment but it's planned to be implement soon.)

Regardless, I'm ok with emitting the event synchronously if you say it's fine.

Reply:
The event would take two event loop cycles to run but I don't think it's a big deal either way. It's up to you.

Reply:
Well, then I'd leave it as is.